### PR TITLE
Update dot.hpp for v3

### PR DESCRIPTION
**Pull request by james-d-mitchell**
Just parking this here for visibility, not ready yet.

### CODE REVIEW

**Comment by james-d-mitchell** [comment]
This is now ready for review @Joseph-Edwards if you'd be so kind.

File: include/libsemigroups/dot.hpp [hunk 1]
```diff
@@ -61,17 +63,100 @@ namespace libsemigroups {
...
+  //!
+  //! The class Dot facilitates the creation and rendering of graph
+  //! descriptions in the [DOT][] language of the [Graphviz][] graph drawing
+  //! software.
+  //!
+  //! The function \ref dot can be used to produce initial Dot objects for
+  //! various of the data structures and algorithms in ``libsemigroups``:
+  //!
+  //! * \ref dot(AhoCorasick&)
+  //! * \ref dot(WordGraph<Node> const& wg)
```
Comment (Joseph-Edwards) [inline include/libsemigroups/dot.hpp:81]: This doesn't seem to create a link for me, though I can't see why not. Would you expect it to?

**Comment by Joseph-Edwards** [comment]
Other than the link to `WordGraph`, this all looks good to me!

File: include/libsemigroups/dot.hpp (hunk 1 above)
Comment (james-d-mitchell) [reply include/libsemigroups/dot.hpp:81]: That's intentional at present, there's no such do to link to.

**Pull request merged**